### PR TITLE
Section 9-9: Dexter Chew (Beat That)

### PR DESCRIPTION
**Pull request by dexterch91**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
40min

Please fill in one error and/or error message you received while working on this assignment.
undefined values due to logical mistakes.

What part of the assignment did you spend the most time on?
Thinking about whether to refactor now or go to sleep.

Comfort Level (1-5): 4

Completeness Level (1-5): 2

What did you think of this deliverable?
Ok good enough. But hold up, I never used arrays.

Is there anything in this code that you feel pleased about?
Not much yet. See next commit.

### CODE REVIEW

**Comment by dexterch91** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
1.5 hours

Please fill in one error and/or error message you received while working on this assignment.
undefined values due to logical mistakes.
A ton of errors mainly logical errors and synatx which are solvable.

What part of the assignment did you spend the most time on?
Refactoring for scalability.

Comfort Level (1-5): 4

Completeness Level (1-5): 4

What did you think of this deliverable?
Not bad. The refactoring approach works. Googling for max index works too haha.

Is there anything in this code that you feel pleased about?
Refactoring process.

File: script.js [hunk 1]
```diff
@@ -1,4 +1,114 @@
+var gameState = "Init"; //Game States
+var pTotal = []; //Array storage for totals
+var noPlayers = 5; //Placeholder for constant
+var roll1 = 0; //Generic roll number 1st
+var roll2 = 0; //Generic roll number 2nd
+var pCount = 1; //Player counter for while loop
+
```
Comment (jokatty) [inline script.js:7]: very clear comments. Keep it up.

File: script.js [hunk 2]
```diff
@@ -1,4 +1,114 @@
...
+  return randInt;
+};
+
+//Function 2: Giving the total number based on order
+var orderSelect = function (diceOne, diceTwo, userSel) {
+  if (userSel == 1) {
+    return Number(diceOne + diceTwo);
+  }
+  if (userSel == 2) {
+    return Number(diceTwo + diceOne);
```
Comment (jokatty) [inline script.js:91]: Good use of `if and return` block

File: script.js [hunk 3]
```diff
@@ -1,4 +1,114 @@
...
+    `Welcome Player ${pArray}. <br><br> You rolled ${p1DiceRoll1} for Dice 1 and ${p1DiceRoll2} for Dice 2. <br><br> Choose the order of the dice.`,
+    p1DiceRoll1,
+    p1DiceRoll2,
+  ];
+};
+
+var recNumber = function (arrayz) {
+  var textMessage = "";
+  var count = 0;
+  while (count < arrayz.length) {
```
Comment (jokatty) [inline script.js:109]: please give a descriptive name to a variable. Name arrayz is not very clear as to what this array holds. Might be confusing when you come back to this code in the future.

File: script.js [hunk 4]
```diff
@@ -1,4 +1,114 @@
...
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = diceGame(input);
   return myOutputValue;
 };
+
+//Function 0: Main Game Alogrithm
+var diceGame = function (userInput) {
+  var message = ""; //generic message
+  while (pCount <= noPlayers) {
```
Comment (jokatty) [inline script.js:16]: pCount? 
please try to give a more descriptive name.

File: script.js (hunk 3 above)
Comment (jokatty) [reply script.js:109]: Overall, really good code quality and logic. Keep it up !

File: script.js [hunk 5]
```diff
@@ -1,4 +1,114 @@
...
+        return `Error! Please choose numeral order by typing in 'Dice 1' or 'Dice 2' as the first numeral.
+        <br>
+        <br>
+        ${recNumber(pTotal)}`;
+      }
+    }
+  }
+};
+
+//Function 1: Geneerating Random Number
```
Comment (jokatty) [inline script.js:78]: good practice commenting on these functions. Keep it up. Also consider using https://jsdoc.app/about-getting-started.html commenting syntax.

**Pull request closed without merging**